src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -104,65 +104,58 @@ rpmostreed_deployment_get_for_index (OstreeSysroot *sysroot,
   return g_object_ref (deployments->pdata[deployment_index]);
 }
 
-static GVariant *
-rpmostreed_deployment_gpg_results (OstreeRepo *repo,
+static gboolean
+rpmostreed_deployment_gpg_results (OstreeRepo  *repo,
                                    const gchar *origin_refspec,
-                                   const gchar *csum,
-                                   gboolean *out_enabled)
+                                   const gchar *checksum,
+                                   GVariant   **out_results,
+                                   gboolean    *out_enabled,
+                                   GError     **error)
 {
-  GError *error = NULL;
-  GVariant *ret = NULL;
+  GLNX_AUTO_PREFIX_ERROR ("GPG verification error", error);
 
   g_autofree gchar *remote = NULL;
-  glnx_unref_object OstreeGpgVerifyResult *result = NULL;
-
-  guint n_sigs, i;
-  gboolean gpg_verify;
-  g_auto(GVariantBuilder) builder;
-
-  g_variant_builder_init (&builder, G_VARIANT_TYPE ("av"));
-
-  if (!ostree_parse_refspec (origin_refspec, &remote, NULL, &error))
-    goto out;
+  if (!ostree_parse_refspec (origin_refspec, &remote, NULL, error))
+    return FALSE;
 
+  gboolean gpg_verify = FALSE;
   if (remote)
     {
-      if (!ostree_repo_remote_get_gpg_verify (repo, remote, &gpg_verify, &error))
-        goto out;
-    }
-  else
-    {
-      gpg_verify = FALSE;
+      if (!ostree_repo_remote_get_gpg_verify (repo, remote, &gpg_verify, error))
+        return FALSE;
     }
 
-  *out_enabled = gpg_verify;
   if (!gpg_verify)
-    goto out;
-
-  result = ostree_repo_verify_commit_for_remote (repo, csum, remote, NULL, &error);
-  if (!result)
-    goto out;
-
-  n_sigs = ostree_gpg_verify_result_count_all (result);
-  if (n_sigs < 1)
-    goto out;
-
-  for (i = 0; i < n_sigs; i++)
     {
-      g_variant_builder_add (&builder, "v",
-                             ostree_gpg_verify_result_get_all (result, i));
+      /* Note early return; no need to verify signatures! */
+      *out_enabled = FALSE;
+      *out_results = NULL;
+      return TRUE;
     }
 
-  ret = g_variant_builder_end (&builder);
+  g_autoptr(GError) local_error = NULL;
+  g_autoptr(OstreeGpgVerifyResult) verify_result =
+    ostree_repo_verify_commit_for_remote (repo, checksum, remote, NULL, &local_error);
+  if (!verify_result)
+    {
+      /* Somehow, we have a deployment which has gpg-verify=true, but *doesn't* have a valid
+       * signature. Let's not just bomb out here. We need to return this in the variant so
+       * that `status` can show "(unsigned)". */
```

Comment:
Well it's not necessarily unsigned, the signature could be invalid right?  Not something to fix in this patch...handling things correctly here feels ugly, since the deployment data is a property and returning an error from reading that doesn't feel right. Maybe the best thing to do is to log to the journal once?  Anyways not something to do in this patch.